src/NiceHashMinerLegacy/Switching/NHSmaData.cs

```diff
@@ -88,20 +78,21 @@ public static void InitializeIfNeeded()
 
         #region Update Methods
 
+        // TODO maybe just swap the dictionaries???
```

Comment:
We'd have to make sure the socket call doesn't exclude an algorithm at some point. It could cause problems if an algorithm is removed from the dict while it is being mined on.

Either way, it would probably still be better to at least copy the dict (`_currentPayingRates = new Dictionary<AlgorithmType, double>(newSma);`). Otherwise, the caller of this func could update the dictionary it passes through and cause havoc with thread safety.